resources/popup.js

```diff
@@ -1,27 +1,30 @@
+// Use the toggle button on the popup to switch the loggingEnabled option.
+function toggle() {
+    chrome.storage.sync.get({ loggingEnabled: true }, toggleSetter);
+}
+
+// Helper function that sets loggingEnabled to the right new value.
+function toggleSetter(settingsObject) {
+    chrome.storage.sync.set({
+        loggingEnabled: !(settingsObject.loggingEnabled)
+    })
+}
+
 $(document).ready(function () {
```

Comment:
I understand your point here. The thing is, if I split the functions, I will break the function, as I will have to save the get callback somewhere..
That's why I tried to process it via this way.